src/background.js

```diff
@@ -11,7 +11,12 @@ class AccessToken {
     tokenPool.generateTokens();
   }
 
-  static get() {
+  static async get() {
+    const token = AccessToken.TOKEN;
+    if (token) {
+      return token;
+    }
+    await AccessToken.refresh();
     return AccessToken.TOKEN;
```

Comment:
I'm not sure this will return the new token after a refresh. `AccessToken.refresh` will resolve after the fetch to the refresh token endpoint has completed, but `AccessToken.TOKEN` is updated after the cookie updated event.